### PR TITLE
Onboarding review

### PR DESCRIPTION
**Pull request by fffergal**
- Add recipes but not ingredients
- Add ingredients
- Update README with guide for playing with API
- Add uWSGI to reject big payloads

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -2,3 +2,4 @@ Django>=3.2.4,<3.3
 djangorestframework>=3.12.4,<3.13
 psycopg2>=2.8.6,<2.9
 drf-spectacular>=0.15.1,<0.16
+uwsgi>=2.0.21,<2.1
```
Comment (joe-gb) [inline requirements.txt:5]: Curious - was this needed over just using runserver?

File: app/recipe/serializers.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        if ingredients:
+            for ingredient in ingredients:
+                Ingredient.objects.create(recipe=recipe, **ingredient)
+        return recipe
+
+    def update(self, instance, validated_data):
+        ingredients = validated_data.pop("ingredients", None)
+        recipe = super().update(instance, validated_data)
+        if ingredients:
+            Ingredient.objects.filter(recipe=recipe).delete()
```
Comment (joe-gb) [inline app/recipe/serializers.py:32]: deleting all the ingredients and re-creating them every time could be doing more work than needed, but for this exercise it's fine :smile:

**Review by joe-gb — APPROVED** [review]
Thanks for opening the PR to make it easier to review! Great example of the on-boarding exercise - DRF used properly and test coverage is great. Good stuff :+1:

File: requirements.txt (hunk 1 above)
Comment (fffergal) [reply requirements.txt:5]: I added uwsgi while doing the security exercise. I figured it was safer to have an application server reject oversized payloads before DRF had a chance to parse it and cause an OOM.

File: app/recipe/serializers.py (hunk 2 above)
Comment (fffergal) [reply app/recipe/serializers.py:32]: More work for the computer, less work for me ;)

**Pull request closed without merging**